### PR TITLE
wpan: add BLE HCI

### PR DESCRIPTION
**Pull request by OueslatiGhaith**
this reexports a forked version of [bluetooth-hci](https://github.com/danielgallagher0/bluetooth-hci) that removes `nb`.
The fork still has the same code size issues, but work is being done to reduce it

### CODE REVIEW

File: embassy-stm32-wpan/src/evt.rs [hunk 1]
```diff
@@ -105,6 +106,14 @@ impl EvtBox {
         Self { ptr }
     }
 
+    pub fn evt<'a>(&self) -> &'a [u8] {
```
Comment (Dirbaio) [inline embassy-stm32-wpan/src/evt.rs:109]: This is unsound. It allows you to hold on to the slice after you've dropped the packet.

You probably want `pub fn evt<'a>(&'a self) -> &'a [u8]`.
Or `pub fn evt(&self) -> &[u8]` which is the same using lifetime elision.

Comment (OueslatiGhaith) [reply embassy-stm32-wpan/src/evt.rs:109]: oh this is unused, forgot to remove it

File: embassy-stm32-wpan/src/evt.rs [hunk 2]
```diff
@@ -124,6 +133,20 @@ impl EvtBox {
             slice::from_raw_parts(p_payload, payload_len as usize)
         }
     }
+
+    /// writes an underlying [`EvtPacket`] into the provided buffer.
+    /// Returns the number of bytes that were written.
+    /// Returns an error if event kind is unknown or if provided buffer size is not enough.
+    pub fn serial<'a>(&self) -> &'a [u8] {
```
[the remark this answers is not in the document]
Comment (OueslatiGhaith) [reply embassy-stm32-wpan/src/evt.rs]: copy on read should solve this, as the result of this operation may live longer that the packet itself

**Comment by xoviat** [comment]
alright, looks good!

**Pull request merged**